sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/MainView.groovy

```diff
@@ -0,0 +1,81 @@
+package life.qbic.portal.sampletracking.components
+
+
+import com.vaadin.ui.HorizontalLayout
+import com.vaadin.ui.Label
+import com.vaadin.ui.VerticalLayout
+import com.vaadin.ui.themes.ValoTheme
+import life.qbic.portal.sampletracking.components.projectoverview.ProjectOverviewView
+import life.qbic.portal.sampletracking.components.sampleoverview.SampleOverviewView
+import life.qbic.portal.sampletracking.components.toggle.ToggleButton
+
+/**
+ * <b><short description></b>
+ *
+ * <p><detailed description></p>
+ *
+ * @since <version tag>
+ */
+class MainView extends VerticalLayout {
+
+    private final ProjectOverviewView projectOverviewView
+    private final SampleOverviewView sampleOverviewView
+    private final HorizontalLayout hotbar = new HorizontalLayout()
+    protected final Label titleLabel = new Label("Sample Status Portlet")
+    private final ToggleButton projectSampleToggle
+
+    MainView(ProjectOverviewView projectOverviewView, SampleOverviewView sampleOverviewView) {
+        this.setMargin(false)
+
+        titleLabel.addStyleName(ValoTheme.LABEL_HUGE)
+
+        this.projectOverviewView = projectOverviewView
+        this.sampleOverviewView = sampleOverviewView
+        addHotbarItem(projectOverviewView.getHotbar())
+        addHotbarItem(sampleOverviewView.getHotbar())
+        showProjectView(true)
+        showSampleView(false)
+
+        projectSampleToggle = setupProjectSampleToggle()
+        addToggleButtonListeners()
+        hotbar.addComponentAsFirst(projectSampleToggle)
+
+        this.addComponents(titleLabel, hotbar, projectOverviewView, sampleOverviewView)
+    }
+
+
+    private void showProjectView(Boolean visible) {
+        projectOverviewView.setVisible(visible)
+        projectOverviewView.getHotbar().setVisible(visible)
+        if (visible) this.titleLabel.value = projectOverviewView.getTitle()
+    }
+
+    private void showSampleView(Boolean visible) {
+        sampleOverviewView.setVisible(visible)
+        sampleOverviewView.getHotbar().setVisible(visible)
+        if (visible) this.titleLabel.value = sampleOverviewView.getTitle()
```

Comment:
I am not sure if we should remove the title or if it would make sense to have an app-title in the main view and add subtitles for the sample view. At least this would make more sense to me.

Reply:
I think the app title is implied by where it is launched/ what is shown. I based the implementation on the current look.